core/src/main/java/com/datastax/dse/driver/internal/core/cql/continuous/ContinuousRequestHandlerBase.java

```diff
@@ -1590,18 +1597,20 @@ private void completeResultSetFuture(
     private ExecutionInfo createExecutionInfo(@NonNull Result result, @Nullable Frame response) {
       ByteBuffer pagingState =
           result instanceof Rows ? ((Rows) result).getMetadata().pagingState : null;
-      return new DefaultExecutionInfo(
-          statement,
-          node,
-          startedSpeculativeExecutionsCount.get(),
-          executionIndex,
-          errors,
-          pagingState,
-          response,
-          true,
-          session,
-          context,
-          executionProfile);
+      this.executionInfo =
```

Comment:
Added an instance variable here for the `ExecutionInfo`.  It gets populated and later passed to the `RequestTracker` in `completeResultSetFuture`.

Reply:
I'm not sure this will always work correctly - the `executionInfo` field on the `NodeResponseCallback` is populated when the response is processed in `processResultResponse`, but before being used when firing the request tracker methods in `completeResultSetFuture`, responses may be queued. If another request comes in before the earlier response is removed from the queue the `executionInfo` field will be overwritten. When the earlier response is later removed from the queue, the execution info provided to the request tracker could be the one from the newer request.

Instead of a field attached to the `NodeResponseCallback`, perhaps you could use the execution info stored on the `resultSet` object in `processResultResponse`? - it's a shame there isn't a common interface (perhaps a useful improvement to file for another time) but it looks like both `ContinuousAsyncResultSet` and `AsyncGraphResultSet` have getters for ExecutionInfo objects.

Reply:
On further thought, since you only have an `Object` at this stage, I don't know how you would get an execution info for the request tracker call if you end up with a something that isn't a `ContinuousAsyncResultSet` or `AsyncGraphResultSet` so perhaps the solution I suggested above isn't going to work.

Reply:
I apologize for calling out details you  no doubt already know off the top of your head, it's just for my own benefit being new to the code.

This _is_ a bit of a pickle.  `ResultSetT` is just an `Object` and it looks like just sharpening up the definition of that type parameter is impossible because there's not a unified type structure for the different kinds of results in the internal API.

I could add a type check and cast for both `AsyncPagingIterable` and `AsyncGraphResultSet` which both provide a `getExecutionInfo` method.

If that works, then we might want to go a bit further and refactor a bit to add a marking interface or something to prevent other types from being passed in, but that seems like an extensive change.

Reply:
It's a bit of work to set up. I ran all the tests against this branch, and it passed all the tests.